firmware/open_evse/J1772EvseController.cpp

```diff
@@ -2091,9 +2090,9 @@ int J1772EVSEController::HsRestoreAmpacity() {
 }
 
 int J1772EVSEController::HsExpirationCheck() {
-  unsigned long sinceLastPulse = (unsigned long)((millis() - m_HsLastPulse)/1000.0); //convert ms to seconds
+  unsigned long sinceLastPulse = millis() - m_HsLastPulse;
   int rc=1;
-  if ((m_HsInterval != 0) && (sinceLastPulse > m_HsInterval)) { //HEARTBEAT_SUPERVISION is currently active and the Heartbeat interval has timed out
+  if (m_HsInterval != 0 && sinceLastPulse > (m_HsInterval * 1000)) { //HEARTBEAT_SUPERVISION is currently active and the Heartbeat interval has timed out
```

Comment:
Definitely double check my work here- by not dividing by 1000 on line 2093, we can multiply the other side of our comparison by 1000 on line 2095 and the logic is equivalent.